runtime/bindings/python/src/pyopenvino/core/infer_request.cpp

```diff
@@ -0,0 +1,284 @@
+            return self._request.get_tensor(name);
+        },
+        py::arg("name"));
+
+    cls.def(
+        "get_tensor",
+        [](InferRequestWrapper& self, const ov::Output<const ov::Node>& port) {
+            return self._request.get_tensor(port);
+        },
+        py::arg("port"));
+
+    cls.def(
+        "get_tensor",
+        [](InferRequestWrapper& self, const ov::Output<ov::Node>& port) {
+            return self._request.get_tensor(port);
+        },
+        py::arg("port"));
+
+    cls.def(
+        "get_input_tensor",
+        [](InferRequestWrapper& self, size_t idx) {
+            return self._request.get_input_tensor(idx);
+        },
+        py::arg("idx"));
+
+    cls.def("get_input_tensor", [](InferRequestWrapper& self) {
+        return self._request.get_input_tensor();
+    });
+
+    cls.def(
+        "get_output_tensor",
+        [](InferRequestWrapper& self, size_t idx) {
+            return self._request.get_output_tensor(idx);
+        },
+        py::arg("idx"));
+
+    cls.def("get_output_tensor", [](InferRequestWrapper& self) {
+        return self._request.get_output_tensor();
+    });
+
+    cls.def(
+        "set_tensor",
+        [](InferRequestWrapper& self, const std::string& name, const ov::runtime::Tensor& tensor) {
+            self._request.set_tensor(name, tensor);
+        },
+        py::arg("name"),
+        py::arg("tensor"));
+
+    cls.def(
+        "set_tensor",
+        [](InferRequestWrapper& self, const ov::Output<const ov::Node>& port, const ov::runtime::Tensor& tensor) {
+            self._request.set_tensor(port, tensor);
+        },
+        py::arg("port"),
+        py::arg("tensor"));
+
+    cls.def(
+        "set_tensor",
+        [](InferRequestWrapper& self, const ov::Output<ov::Node>& port, const ov::runtime::Tensor& tensor) {
+            self._request.set_tensor(port, tensor);
+        },
+        py::arg("port"),
+        py::arg("tensor"));
+
+    cls.def(
+        "set_input_tensor",
+        [](InferRequestWrapper& self, size_t idx, const ov::runtime::Tensor& tensor) {
+            self._request.set_input_tensor(idx, tensor);
+        },
+        py::arg("idx"),
+        py::arg("tensor"));
+
+    cls.def(
+        "set_input_tensor",
+        [](InferRequestWrapper& self, const ov::runtime::Tensor& tensor) {
+            self._request.set_input_tensor(tensor);
+        },
+        py::arg("tensor"));
+
+    cls.def(
+        "set_output_tensor",
+        [](InferRequestWrapper& self, size_t idx, const ov::runtime::Tensor& tensor) {
+            self._request.set_output_tensor(idx, tensor);
+        },
+        py::arg("idx"),
+        py::arg("tensor"));
+
+    cls.def(
+        "set_output_tensor",
+        [](InferRequestWrapper& self, const ov::runtime::Tensor& tensor) {
+            self._request.set_output_tensor(tensor);
+        },
+        py::arg("tensor"));
+
+    cls.def_property_readonly("inputs", [](InferRequestWrapper& self) {
```

Comment:
I don't see such API in C++

Reply:
@jiwaszki ping here

Reply:
@ilyachur We aligned as much as we were able to, sometimes there are some limits on Python side, sometimes due to on-going development of new C++ API.
However I know that we have some independence for Python API development - this method is just a helper to simplify access for the Python users. One of the example is inside tests runtime code (here for outputs):
https://github.com/openvinotoolkit/openvino/blob/3621e39f293d2daf3cf65a8a1bdb901959dce11a/runtime/bindings/python/tests/runtime.py#L138
I do not see how it affects alignment in a wrong way, it only provides user-friendly way to access data.